src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java

```diff
@@ -570,8 +574,11 @@ public Builder retentionStrategy(RetentionStrategy retentionStrategy) {
             return this;
         }
 
+        @SuppressFBWarnings(value = "UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR", justification = "Warning is for the cloud field, and a null "
+                                                                                                + "validation check is done before using it.")
         private RetentionStrategy determineRetentionStrategy() {
```

Comment:
Refactoring to `private static RetentionStrategy determineRetentionStrategy(@NonNull KubernetesCloud cloud, @NonNull PodTemplate podTemplate)` would avoid these extra checks and suppress warnings

Reply:
Good idea indeed, I fixed it.